### PR TITLE
feat: add upcoming auction results screen

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [CX-3227] 

### Description

This PR does the following things: 
- Add the upcoming auction results screen
- Refactor the existing auction results screen and create a screen wrapper that can return Past, Upcoming and All auction results depending on the state provided
- Allow deep linking the route in Android (in iOS works by default)

https://user-images.githubusercontent.com/11945712/208940101-f8cccf09-cfa6-4c24-abb5-636eaa0d9d7d.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add upcoming auction results screen - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3227]: https://artsyproduct.atlassian.net/browse/CX-3227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MrSltun — APPROVED** [review]
LGTM! Great work Mounir!! 🌟

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -117,6 +117,7 @@
                 <data android:pathPrefix="/show" />
                 <data android:pathPrefix="/user/conversations" />
                 <data android:pathPrefix="/user/purchases" />
+                <data android:pathPrefix="/upcoming-auction-results" />
```
Comment (dariakoko) [inline android/app/src/main/AndroidManifest.xml:120]: **Q:** is it for deep links?

**Review by dariakoko — APPROVED** [review]
Looks great ⭐ 
I only left one question

**Review by egdbear — APPROVED** [review]
🚀 
Curious for the order, we show Upcoming Auctions UX desicion? :)

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (MounirDhahri) [reply android/app/src/main/AndroidManifest.xml:120]: Yes, for Android we need to explicitly add the route to `AndroidManifest.xml` to enable deep linking for a route

Comment (MounirDhahri) [reply android/app/src/main/AndroidManifest.xml:120]: Usually this is not needed but our infra is based on explicitly specifying the routes that's why we needed to add it here

**Pull request merged**